src/js/tech/tech.js

```diff
@@ -805,6 +806,12 @@ Tech.withSourceHandlers = function(_Tech){
     this.disposeSourceHandler();
     this.off('dispose', this.disposeSourceHandler);
 
+    // if we have a source and get another one
```

Comment:
what if you have a tech that isn't a source handler get a new source? Will the these get cleared out then?